Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -0,0 +1,133 @@
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.uri)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: ["+propIVarName + " absoluteString] forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .string(format: .some(.dateTime)):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: [[NSValueTransformer valueTransformerForName:\(dateValueTransformerKey)] transformedValue:\(propIVarName)]  forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .enumT(.integer(_)):
+            return "[dict setObject: [NSNumber numberWithInteger:"+propIVarName + "] forKey: @\"" + param + "\" ];"
+        case .enumT(.string(_)):
+            return "[dict setObject: "+enumToStringMethodName(propertyName: param, className: self.className) + "(\(propIVarName))" + " forKey: @\"" + param + "\" ];"
+        case .array(itemType: _):            
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+
+        case .map(valueType: _):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .oneOf(types: _):
+            return ObjCIR.scope {[
+                ObjCIR.ifStmt("\(propIVarName) != nil") {[
+                    "[dict setObject: "+propIVarName + " forKey: @\"" + param + "\" ];"
+                    ]},
+                ObjCIR.elseStmt({[
+                    "[dict setObject: [NSNull null] forKey: @\"" + param + "\" ];"
+                    ]
+                })
+                ]}
+        case .reference(with: _):
```

Comment:
A `reference` Schema type will contain a closure that returns a Schema. This is only expected to be an object for now but you can capture this closure in a variable in the `case` statement. You can then return a recursive call to this method with that variable which will be another Schema type besides `reference`